proto/minder/v1/minder.proto

```diff
@@ -3411,3 +3412,139 @@ message UpstreamEntityRef {
     // be stored in Minder.
     google.protobuf.Struct properties = 3;
 }
+
+// DataSource is a Data source instance. Data sources represent
+// external integrations that enrich the data in Minder, but do not
+// have explicit lifecycle objects (entities).  Integrations which
+// create entities are called Providers.
+message DataSource {
+    // version is the version of the data source API.
+    string version = 1 [
+        (buf.validate.field).string = {
+            pattern: "^v\\d$",
+        },
+        (google.api.field_behavior) = REQUIRED
+    ];
+
+    // type is the data source type
+    string type = 2 [
+        (buf.validate.field).string = {
+            in: ["data-source"],
+            max_len: 12,
+        }
+    ];
+
+    // context is the context in which the data source is evaluated.
+    // Note that in this case we only need the project in the
+    // context, since data sources are not provider-specific.
+    ContextV2 context = 3;
+
+    // name is the name of the data source.
+    // Note that this is unique within a project hierarchy.
+    // This is also case insensitive.
+    string name = 4 [
+        (buf.validate.field).string = {
+            pattern: "^[A-Za-z][-[:word:]]*$",
+            max_len: 200,
+        },
+        (google.api.field_behavior) = REQUIRED
+    ];
+
+    // id is the unique identifier of the data source.
+    string id = 5 [
+        (buf.validate.field).string = {uuid: true},
+        (google.api.field_behavior) = OUTPUT_ONLY
+    ];
+
+    oneof driver {
+        // rest is the REST data source driver.
+        RestDataSource rests = 6;
```

Comment:
@blkt note that I'm not able to use a `map` in a `oneof` in protobuf [1]. So I needed to nest the map within the `RestDataSource` message.

[1] https://github.com/protocolbuffers/protobuf/issues/7542

Reply:
**nit:** sorry for bringing this up so late, but given that we don't have a `rest` field anymore, I would rename `rests` into `rest`.